### PR TITLE
Update rules' sync triggers.

### PR DESCRIPTION
**Pull request by orendain**
Resolves https://github.com/orendain/DemeoMods/issues/261. ... I hope.

**Changes:**
- Rename `SyncableTrigger` enums and corresponding rule field name.
- Document each `SyncableTrigger`.
- Modify rules' `SyncableTrigger`s.

~> Note: I still need to playtest and see if things are working as intended.~

Update:  Tests went well.  Played a full game of Arachnophobia with no sync issues (from what I could gather from the group) and also some very specific testing of `PieceUseWhenKilledOverrideRule` and `AbilityHealOverriddenRule` with the help of some random players.

### CODE REVIEW

File: HouseRules_Essentials/Rules/PieceUseWhenKilledOverriddenRule.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public sealed class PieceUseWhenKilledOverriddenRule : Rule, IConfigWritable<Dic
     {
         public override string Description => "Piece UseWhenKilled lists are overridden";
 
-        protected override SpecialSyncData ModifiedData => SpecialSyncData.PieceData;
+        protected override SyncableTrigger ModifiedSyncables => SyncableTrigger.NewPieceModified | SyncableTrigger.StatusEffectDataModified;
```
Comment (orendain) [inline HouseRules_Essentials/Rules/PieceUseWhenKilledOverriddenRule.cs:15]: Also added `SyncableTrigger.StatusEffectDataModified` to the mix, as this rule could be used to trigger an ability like `Freeze`, that will cause some update to status effects.

File: HouseRules_Essentials/Rules/AbilityHealOverriddenRule.cs [hunk 2]
```diff
@@ -12,8 +12,6 @@ public sealed class AbilityHealOverriddenRule : Rule, IConfigWritable<Dictionary
     {
         public override string Description => "Ability Heal settings are adjusted";
 
-        protected override SpecialSyncData ModifiedData => SpecialSyncData.StatusEffectData;
```
[the remark this answers is not in the document]
Comment (orendain) [reply HouseRules_Essentials/Rules/AbilityHealOverriddenRule.cs:15]: Update:  Tested this rule without this declared and results were successful.  Clients saw animations with the typical heal amounts, but the game auto-synced with the correct heal amounts (the game autosyncs damage and particular stats, as we've previously found out).

**Review by orendain — COMMENTED** [review]
Added comments on changes below.

**Review by jimconner — APPROVED** [review]
All builds cleanly, and no errors in logs after a play-through. Merging.

**Pull request merged**